### PR TITLE
SNOW-1488913: Implement `Resampler.first` and `Resampler.last`

### PR DESCRIPTION
**Pull request by sfc-gh-nkrishna**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1488913

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: src/snowflake/snowpark/modin/plugin/_internal/resample_utils.py [hunk 1]
```diff
@@ -51,6 +51,8 @@
     "var",
     "count",
     "size",
+    "first",
+    "last",
```
Comment (sfc-gh-nkrishna) [inline src/snowflake/snowpark/modin/plugin/_internal/resample_utils.py:55]: Adding to list of implemented Resampler APIs ensures the tests for these API run.

File: src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py [hunk 2]
```diff
@@ -9718,6 +9718,17 @@ def resample(
                     )
                     .set_index_names([None])
                 )
+            elif resample_method in ("first", "last"):
+                # Call groupby_first or groupby_last directly
+                qc = getattr(
+                    SnowflakeQueryCompiler(frame), "groupby_" + resample_method
```
Comment (sfc-gh-dpetersohn) [inline src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py:9724]: f-strings are preferred to `+`
```suggestion
                    SnowflakeQueryCompiler(frame), f"groupby_{resample_method}"
```

**Review by sfc-gh-dpetersohn — COMMENTED** [review]
Looks great, left a suggestion.

**Review by sfc-gh-lmukhopadhyay — APPROVED** [review]
Lgtm!

**Review by sfc-gh-dpetersohn — APPROVED** [review]

**Pull request merged**